### PR TITLE
pybind/mgr/rbd_support: ignore missing support for RBD namespaces

### PR DESCRIPTION
**Pull request by dillaman**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/rbd_support/module.py [hunk 1]
```diff
@@ -673,9 +673,16 @@ def init_task_queue(self):
...
-                        self.load_task_queue(ioctx, pool_name)
+                    try:
+                        namespaces = rbd.RBD().namespace_list(ioctx)
+                    except rbd.OperationNotSupported:
+                        self.log.debug("Namespaces not supported")
+                        continue
+
+                    for namespace in namespaces:
+                            ioctx.set_namespace(namespace)
+                            self.load_task_queue(ioctx, pool_name)
```
Comment (trociny) [inline src/pybind/mgr/rbd_support/module.py]: nit: why 8 spaces indentation?

Comment (dillaman) [reply src/pybind/mgr/rbd_support/module.py]: Good catch, I had put a `try` around the whole block but then didn't like it.

**Comment by trociny** [comment]
@batrick Did you have a chance to test it already? I would rather like this to be backported before the forthcoming nautilus release.

**Comment by batrick** [comment]
> @batrick Did you have a chance to test it already? I would rather like this to be backported before the forthcoming nautilus release.

Yes, sorry for not getting back to you both. There were other failures in the fs:upgrade suite that were obscuring the results.

It seems though that this fix does work as advertised. I see these messages though:

````
pdonnell@teuthology /a/pdonnell-2019-08-15_13:19:31-fs-wip-pdonnell-testing-20190814.222632-distro-basic-smithi$ grep librbd::api::Namespace: */teu*
4216926/teuthology.log:2019-08-15T14:56:34.020 INFO:tasks.ceph.mgr.y.smithi172.stderr:2019-08-15T14:56:34.016+0000 7ff2e57d9700 -1 librbd::api::Namespace: list: error listing namespaces: (95) Operation not supported
4216958/teuthology.log:2019-08-15T15:03:09.795 INFO:tasks.ceph.mgr.y.smithi037.stderr:2019-08-15T15:03:09.791+0000 7f1090d79700 -1 librbd::api::Namespace: list: error listing namespaces: (95) Operation not supported
4216996/teuthology.log:2019-08-15T15:03:11.151 INFO:tasks.ceph.mgr.x.smithi093.stderr:2019-08-15T15:03:11.143+0000 7f15f9316700 -1 librbd::api::Namespace: list: error listing namespaces: (95) Operation not supported
4217035/teuthology.log:2019-08-15T16:49:13.076 INFO:tasks.ceph.mgr.y.smithi060.stderr:2019-08-15T16:49:13.070+0000 7f0156cc0700 -1 librbd::api::Namespace: list: error listing namespaces: (95) Operation not supported
````

From: /ceph/teuthology-archive/pdonnell-2019-08-15_13:19:31-fs-wip-pdonnell-testing-20190814.222632-distro-basic-smithi


If that's expected, then cool. It doesn't appear to break the upgrade process.

**Comment by trociny** [comment]
Yes, the error messages are expected. Thanks!

**Pull request merged**